StackCalculator/Program.cs

```diff
@@ -0,0 +1,238 @@
+﻿/// <summary>
+/// Interface for the classic stack.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public interface IStack<T>
+{
+    /// <summary>
+    /// Pushes a value onto the stack.
+    /// </summary>
+    /// <param name="value">The value to push onto the stack.</param>
+    public void Push(T value);
+
+    /// <summary>
+    /// Pops the top value off the stack and returns it.
+    /// </summary>
+    /// <returns>The top value of the stack.</returns>
+    public T Pop();
+
+    /// <summary>
+    /// Getter for stack size.
+    /// </summary>
+    /// <returns>Stack size.</returns>
+    public int Size { get; }
+}
+
+/// <summary>
+/// Implementation of a stack using a list.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public class StackOnList<T> : IStack<T>
+{
+    private int size;
+    private readonly List<T> list;
+
+    /// <summary>
+    /// StackOnList constructor.
+    /// </summary>
+    public StackOnList()
+    {
+        size = 0;
+        list = new List<T>();
+    }
+    public void Push(T value)
+    {
+        list.Add(value);
+        size++;
```

Comment:
К полям класса стоит обращаться с использованием `this`:
```suggestion
        this.list.Add(value);
        this.size++;
```
И такое стоит сделать везде